Mostly sync BlasKernel.cpp with ATen ReducedPrecisionGemvFastPathKernel

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #10868

The two files were similar, but diverged due to recent changes. Since we have sharing of PyTorch headers, we can keep them mostly the same; differences are some of the namespace stuff, lintrunner, and a couple of EXECUTORCH NOTEs.

Differential Revision: [D74702689](https://our.internmc.facebook.com/intern/diff/D74702689/)

swolchok committed:
Mostly sync BlasKernel.cpp with ATen ReducedPrecisionGemvFastPathKernel

The two files were similar, but diverged due to recent changes. Since we have sharing of PyTorch headers, we can keep them mostly the same; differences are some of the namespace stuff and a couple of EXECUTORCH NOTEs.

Differential Revision: [D74702689](https://our.internmc.facebook.com/intern/diff/D74702689/)

[ghstack-poisoned]

kimishpatel reviewed (approved):
bunch of ci failures. not sure if they are related. can we ask the person who filed the rpi issue to see if this fixes it?

swolchok commented:
/pytorch/executorch/kernels/optimized/blas/BlasKernel.cpp:17:10: fatal error: 'ATen/cpu/vec/functional.h' file not found

swolchok committed:
Update on "Mostly sync BlasKernel.cpp with ATen ReducedPrecisionGemvFastPathKernel"


The two files were similar, but diverged due to recent changes. Since we have sharing of PyTorch headers, we can keep them mostly the same; differences are some of the namespace stuff and a couple of EXECUTORCH NOTEs.

Differential Revision: [D74702689](https://our.internmc.facebook.com/intern/diff/D74702689/)

[ghstack-poisoned]

swolchok committed:
fix CMake on "Mostly sync BlasKernel.cpp with ATen ReducedPrecisionGemvFastPathKernel"


The two files were similar, but diverged due to recent changes. Since we have sharing of PyTorch headers, we can keep them mostly the same; differences are some of the namespace stuff and a couple of EXECUTORCH NOTEs.

Differential Revision: [D74702689](https://our.internmc.facebook.com/intern/diff/D74702689/)

[ghstack-poisoned]

swolchok committed:
lintrunner on "Mostly sync BlasKernel.cpp with ATen ReducedPrecisionGemvFastPathKernel"


The two files were similar, but diverged due to recent changes. Since we have sharing of PyTorch headers, we can keep them mostly the same; differences are some of the namespace stuff and a couple of EXECUTORCH NOTEs.

Differential Revision: [D74702689](https://our.internmc.facebook.com/intern/diff/D74702689/)

[ghstack-poisoned]

swolchok committed:
also enable for non-aarch64 on "Mostly sync BlasKernel.cpp with ATen ReducedPrecisionGemvFastPathKernel"


The two files were similar, but diverged due to recent changes. Since we have sharing of PyTorch headers, we can keep them mostly the same; differences are some of the namespace stuff, lintrunner, and a couple of EXECUTORCH NOTEs.

Differential Revision: [D74702689](https://our.internmc.facebook.com/intern/diff/D74702689/)

[ghstack-poisoned]
